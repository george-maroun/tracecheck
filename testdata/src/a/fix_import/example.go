package fix_import

import (
	"github.com/go-logr/zapr"
	"go.uber.org/zap"
	"context"
)

func SomeFunc(ctx context.Context, eventType, deliveryID string, payload []byte) error {
	log := zapr.NewLogger(zap.L()).WithValues("eventType", eventType, "deliverID", deliveryID)  // want `missing traceId in logging keys`
	log = log.WithValues("eventType", "hello")
	log.Info("Tracing")
	return nil
}