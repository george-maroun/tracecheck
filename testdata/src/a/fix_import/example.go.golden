package fix_import

import (
	"github.com/go-logr/zapr"
	"go.uber.org/zap"
	"context"
)

func SomeFunc(ctx context.Context, eventType, deliveryID string, payload []byte) error {
	log := zapr.NewLogger(zap.L()).WithValues("eventType", eventType, "deliverID", deliveryID)  // want `missing traceId in logging keys`
	log = log.WithValues("eventType", "hello")
	log.Info("Tracing")
	return nil
}

func SomeFunc1(ctx context.Context, eventType, deliveryID string, payload []byte) error {
	const traceLogKey = "dummyTraceId"
	log := zapr.NewLogger(zap.L()).WithValues(traceLogKey, "someValue") // cannot be detected
	log = log.WithValues("eventType", "hello")
	log.Info("Tracing")
	return nil
}

func SomeFunc2(eventType, deliveryID string, payload []byte) error {
	log := zapr.NewLogger(zap.L()).WithValues("eventType", eventType, "deliverID", deliveryID) // cannot be detected
	log = log.WithValues("eventType", "hello")
	log.Info("Tracing")
	return nil
}